### PR TITLE
Use --oss-only for ci tests

### PR DESCRIPTION
**Pull request by aryx**
This is a partial revert of https://github.com/semgrep/semgrep/pull/9872
We do not want to depend on the presence of a semgrep-core-proprietary binary in the e2e tests in the OSS repo, especially now that we merged the OSS and PRO repo.
This was annoying because suddently 'make test' started to fail locally on my machine because I didn't have semgrep-pro installed, which I think is confusing.

### CODE REVIEW

**Comment by aryx** [comment]
This is a partial revert of https://github.com/semgrep/semgrep/pull/9872
We do not want to depend on the presence of a semgrep-core-proprietary binary in the e2e tests in the OSS repo, especially now that we merged the OSS and PRO repo.
This was annoying because suddently 'make test' started to fail locally on my machine because I didn't have semgrep-pro installed, which I think is confusing.

File: cli/tests/default/e2e-other/test_taint_intrafile.py [hunk 1]
```diff
@@ -1,61 +0,0 @@
-import pytest
```
Comment (aryx) [inline cli/tests/default/e2e-other/test_taint_intrafile.py:1]: That's the only test that actually depend on semgrep-core-proprietary, but this test should be moved to the Pro repo!

**Review by IagoAbal — APPROVED** [review]
stamp

OK so we need an equivalent workflow in Pro to test with Pro (intra- or inter-file). cc @amchiclet @emjin

**Comment by IagoAbal** [comment]
> This is a partial revert of #9872 We do not want to depend on the presence of a semgrep-core-proprietary binary in the e2e tests in the OSS repo, especially now that we merged the OSS and PRO repo. This was annoying because suddently 'make test' started to fail locally on my machine because I didn't have semgrep-pro installed, which I think is confusing.

This could be a good PR description, it's better if the reason for the revert is recorded in the Git history.

File: cli/tests/default/e2e-other/test_taint_intrafile.py (hunk 1 above)
Comment (akuhlens) [reply cli/tests/default/e2e-other/test_taint_intrafile.py:1]: Can you move this test to the pro repo, so that we don't drop the ball on migrating it?

**Review by akuhlens — APPROVED** [review]
🙈

**Pull request merged**